### ISSUE TITLE
quantamagazine Website breaks Just Read (and Just Read breaks it)

### ISSUE THREAD

**Issue by Avery2** [ZachSaucier/Just-Read#375]
**Describe the bug**

The article below (and I think any article on that site) doesn't work with Just Read. Once you toggle Just Read, it just freezes the page. Toggling Just Read again doesn't do anything, you must refresh the page to fix it.

https://www.quantamagazine.org/how-bells-theorem-proved-spooky-action-at-a-distance-is-real-20210720/

**To Reproduce**
1. Go to 'https://www.quantamagazine.org/how-bells-theorem-proved-spooky-action-at-a-distance-is-real-20210720/
' (or any article on that site I believe)
2. Click on (enable) 'the Just Read extension'
3. See error: The page is frozen

**Expected behavior**
It should just open Just Read like normal. Or not break freeze the page and let me know Just Read won't work (or even throw an error, I couldn't find any thrown).

**Screenshots**
None

**Please complete the following information:**
 - OS: macOS Big Sur Version 11.4
 - Browser Chrome
 - Version Version 91.0.4472.114 (Official Build) (x86_64)

**Additional context**
<details>
<summary>Pretty sure I'm just using default styles. Styles:</summary>

```
@import url("https://fonts.googleapis.com/css?family=Merriweather|Source+Sans+Pro:400,700");
body { font-family: Merriweather, serif; background-color: rgb(255, 255, 255); line-height: 1.6; font-size: 1.5rem; color: rgb(51, 51, 51); text-rendering: optimizelegibility; }
h1, h2 { font-family: "Source Sans Pro", sans-serif; font-weight: 700; }
h1 { font-size: 1.875em; line-height: 1.125; }
h2 { font-size: 1.45em; line-height: 1.2625em; }
h3 { font-family: Merriweather, serif; font-size: 1.25em; line-height: 1.5; }
hr { height: 1px; background-color: rgb(216, 216, 216); border: none; width: 100%; margin: 0px; }
a[href] { color: rgb(39, 170, 225); }
a[href]:hover { color: rgb(78, 202, 255); }
.simple-container { -webkit-print-color-adjust: exact; }
.youtubeContainer { position: relative; width: 100%; height: 0px; padding-bottom: 56.25%; padding-top: 25px; }
iframe { width: 100%; height: 300px; }
iframe[src*="youtube.com/embed/"] { width: 100%; height: 100%; position: absolute; top: 0px; left: 0px; }
video, img { max-width: 100%; }
li { line-height: 1.5em; margin-bottom: 5px; }
td { border: 1px solid black; padding: 3px 7px; }
pre { padding: 10px; overflow: auto; }
pre, code { background-color: rgb(224, 224, 224); }
blockquote { border-left: 4px solid; padding: 15px 10% 15px 8%; margin: 1em 0px; font-size: 1.2em; line-height: 1.4; }
blockquote > :first-child { margin-top: 0px; }
blockquote > :last-child { margin-bottom: 0px; }
figure { margin: 0px 0px 10px; }
figcaption { font-size: 0.9em; opacity: 0.7; border: 1px solid rgb(238, 238, 238); padding: 17px 17px 12px; }
aside, [class*="sidebar"], [id*="sidebar"] { max-width: 90%; margin: 0px auto; border: 1px solid lightgrey; padding: 5px 15px; }
.simple-date { display: inline-block; font-family: "Source Sans Pro", sans-serif; font-size: 18px; padding-right: 15px; padding-top: 10px; padding-bottom: 10px; border-right: 1px solid rgb(216, 216, 216); }
.rtl .simple-date { border-left: 1px solid rgb(216, 216, 216); border-right: none; padding-right: 0px; padding-left: 15px; }
.simple-author { display: inline-block; font-family: "Source Sans Pro", sans-serif; font-size: 18px; color: rgb(39, 170, 225); line-height: 22px; padding-left: 10px; padding-top: 10px; padding-bottom: 10px; }
.rtl .simple-author { padding-left: 0px; padding-right: 10px; }
.simple-article-container { max-width: 80%; }
[class*="emoji"], [class*="icon"]:not([class*="no-icon"]) { width: 1em; }
[class*="inline"] { display: inline-block; min-width: 2rem; }
```
</details>

**Comment by knudvaneeden** [comment]
Hello Zach,

This issue of breaking the web page is still present. Tried and tested with the latest reinstalled Just Read plugin version 5.1.10 in a Google Chrome browser with latest update on Microsoft Windows 10 Professional.

See e.g. this web page
https://www.quantamagazine.org/new-proof-reveals-the-hidden-structure-of-common-equations-20220421/
where it reads 

"Just two weeks later, in an online [presentation](https://mca2021.dm.uba.ar/en/tools/view-abstract?code=3773) during the [Mathematical Congress of the Americas](https://mca2021.dm.uba.ar/en/), Bhargava shared new work proving van der Waerden’s conjecture in its entirety. He posted his [paper](https://arxiv.org/abs/2111.06507) online in November."

where it breaks after a picture with image address
https://d2r55xnwy6nx47.cloudfront.net/uploads/2017/04/MB_header_web1.jpg

Thanks
with friendly greetings
Knud van Eeden

**Comment by ZachSaucier** [comment]
Hey @knudvaneeden. Thanks for posting! That's actually an instance of Just Read's selection algorithm not being perfect. The open issue related to that is [this one](https://github.com/ZachSaucier/Just-Read/issues/331). 

To work around the issue, you can use Just Read's [user selection mode](https://github.com/ZachSaucier/Just-Read#user-selection-mode).

**Comment by knudvaneeden** [comment]
Hello Zach,
Yes, the workaround looks good. 
Thankswith friendly greetingsKnud van Eeden
    On Thursday, April 21, 2022, 11:22:12 PM GMT+2, Zach Saucier ***@***.***> wrote:  
 
 


Hey @knudvaneeden. Thanks for posting! That's actually an instance of Just Read's selection algorithm not being perfect. The open issue related to that is this one.

To work around the issue, you can use Just Read's user selection mode.

**Comment by knudvaneeden** [comment]
Hello Zach,
https://www.quantamagazine.org/impossible-seeming-surfaces-confirmed-decades-after-conjecture-20220602/

Here even this 'User selection' mode seems to fail.
A large chunk of text not shown even if selecting it and then right click > 'View this selection in JustRead'.



    On Friday, April 22, 2022, 12:23:41 AM GMT+2, knud van eeden ***@***.***> wrote:  
 
  Hello Zach,
Yes, the workaround looks good. 
Thankswith friendly greetingsKnud van Eeden
    On Thursday, April 21, 2022, 11:22:12 PM GMT+2, Zach Saucier ***@***.***> wrote:  
 
 


Hey @knudvaneeden. Thanks for posting! That's actually an instance of Just Read's selection algorithm not being perfect. The open issue related to that is this one.

To work around the issue, you can use Just Read's user selection mode.

**Comment by ZachSaucier** [comment]
Hmm. It seems to work just fine for me when I just click the Just Read button on that page but I could be missing something. What parts of text are not showing for you?

**Comment by knudvaneeden** [comment]
Hello Zach,
https://www.quantamagazine.org/impossible-seeming-surfaces-confirmed-decades-after-conjecture-20220602/

1. You see thus first the original page attached, and below what I see when I click just read page attached.That is different text.
2. Also if I highlight select the first part of text, then right click and try to read it then it jumps again to the second part of the text and does unexpectedly not show that first text.









    On Wednesday, June 8, 2022, 09:36:47 PM GMT+2, Zach Saucier ***@***.***> wrote:  
 
 


Hmm. It seems to work just fine for me when I just click the Just Read button on that page but I could be missing something. What parts of text are not showing for you?

**Comment by ZachSaucier** [comment]
> You see thus first the original page attached, and below what I see when I click just read page attached.That is different text.

Sorry, your attachments didn't get added. Maybe it'd help if you posted directly to [the GitHub issue](https://github.com/ZachSaucier/Just-Read/issues/375) instead?

**Comment by knudvaneeden** [comment]
Hello Zach,
https://www.quantamagazine.org/impossible-seeming-surfaces-confirmed-decades-after-conjecture-20220602/
1. You see thus first the original page attached, and below what I see when I click just read page attached.That is different text.
https://drive.google.com/file/d/1dpfwT8Z80FF3D4L1AaEksUiNmPOGX8dC/view?usp=sharing
https://drive.google.com/file/d/1a1lrakHDMIlenpa8D1fYDLkCb_on3ZVI/view?usp=sharing
===
2. Also if I highlight select the first part of text, then right click and try to read it then it jumps again to the second part of the text and does unexpectedly not show that first text.
https://drive.google.com/file/d/1iZIqF2ksgAlU-6uRabG4ooUmZQ7Fn39l/view?usp=sharing
https://drive.google.com/file/d/1BariMaa49CtjsmjafDOBtz927aKVN23a/view?usp=sharing

    On Wednesday, June 8, 2022, 09:49:15 PM GMT+2, Zach Saucier ***@***.***> wrote:  
 
 



You see thus first the original page attached, and below what I see when I click just read page attached.That is different text.


Sorry, your attachments didn't get added. Maybe it'd help if you posted directly to the GitHub issue instead?

**Comment by knudvaneeden** [comment]
Hello Zach,
It looks like a general pattern that QuantaMagazine breaks Just Read.
Just compare the real text with its Just Read text:
https://www.quantamagazine.org/researchers-achieve-absurdly-fast-algorithm-for-network-flow-20220608/?jr=on

    On Thursday, June 9, 2022, 12:09:52 AM GMT+2, knud van eeden ***@***.***> wrote:  
 
  Hello Zach,
https://www.quantamagazine.org/impossible-seeming-surfaces-confirmed-decades-after-conjecture-20220602/
1. You see thus first the original page attached, and below what I see when I click just read page attached.That is different text.
https://drive.google.com/file/d/1dpfwT8Z80FF3D4L1AaEksUiNmPOGX8dC/view?usp=sharing
https://drive.google.com/file/d/1a1lrakHDMIlenpa8D1fYDLkCb_on3ZVI/view?usp=sharing
===
2. Also if I highlight select the first part of text, then right click and try to read it then it jumps again to the second part of the text and does unexpectedly not show that first text.
https://drive.google.com/file/d/1iZIqF2ksgAlU-6uRabG4ooUmZQ7Fn39l/view?usp=sharing
https://drive.google.com/file/d/1BariMaa49CtjsmjafDOBtz927aKVN23a/view?usp=sharing

    On Wednesday, June 8, 2022, 09:49:15 PM GMT+2, Zach Saucier ***@***.***> wrote:  
 
 



You see thus first the original page attached, and below what I see when I click just read page attached.That is different text.


Sorry, your attachments didn't get added. Maybe it'd help if you posted directly to the GitHub issue instead?

**Comment by ZachSaucier** [comment]
Hey knudvaneeden. There is indeed an issue with Just Read's user selection
mode. I will investigate it more deeply when I get some time. Thanks a lot
for bringing it to my attention!

**Comment by ZachSaucier** [comment]
Hey knudvaneeden. This should be fixed as of Just Read version 5.2.4

**Comment by knudvaneeden** [comment]
OK, thanks Zach.
Currently using Just Read version 5.2.8, thus that should be doing OK.
with friendly greetingsKnud van Eeden
    On Tuesday, October 11, 2022 at 09:48:16 AM GMT+2, Zach Saucier ***@***.***> wrote:  
 
 

Hey knudvaneeden. This should be fixed as of Just Read version 5.2.4
